app/controllers/annotation_categories_controller.rb

```diff
@@ -88,27 +92,36 @@ def create_annotation_text
       last_editor_id: current_user.id
     )
 
+    @assignment = Assignment.find(params[:assignment_id])
+    @annotation_category = @annotation_text.annotation_category
     if @annotation_text.save
       flash_now(:success, t('annotation_categories.update.success'))
-      @assignment = Assignment.find(params[:assignment_id])
-      @annotation_category = @annotation_text.annotation_category
       render :insert_new_annotation_text
     else
-      respond_with @annotation_text, render: { body: nil, status: :bad_request }
+      flash_message(:error, t('.error'))
+      render 'show', assignment_id: @assignment.id, id: @annotation_category.id, status: :bad_request
     end
   end
 
   def destroy_annotation_text
     @annotation_text = AnnotationText.find(params[:id])
     if @annotation_text.destroy
       flash_now(:success, t('.success'))
+    else
+      flash_message(:error, t('.error'))
+      render 'show', assignment_id: @assignment.id, id: @annotation_category.id, status: :bad_request
```

Comment:
These instance variables aren't defined. But also, it's not clear to me that rendering `show` is the best option in this case. We can add a flash message though.